etna/ciim/utils.py

```diff
@@ -175,3 +177,43 @@ def convert_sort_key_to_index(sort):
 
     # Ensure index is always > -1 to prevent invalid offsets being sent to Kong
     return max(index, 0)
+
+
+def get_date_for_retry_after_header(datetime_iso: str) -> str:
+    """
+    datetime_iso expected format "2011-11-04T00:05:23+04:00"
+    Returns formatted date as
+    Example: "<day-name>, <day> <month-name> <year> <hour>:<minute>:<second> GMT"
+             "Wed, 21 Oct 2015 07:28:00 GMT"
+    https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date
+    """
+    d = datetime.datetime.fromisoformat(datetime_iso)
+    day_name = d.strftime("%A")[:3]
+    day = f"{d.day:02}"
+    month_name = d.strftime("%b")
+    year = d.year
+    hour = f"{d.hour:02}"
+    minute = f"{d.minute:02}"
+    second = f"{d.second:02}"
+    return f"{day_name}, {day} {month_name} {year} {hour}:{minute}:{second} GMT"
+
+
+def prevent_request_warnings(original_function):
+    """
+    If we need to test for 503s this decorator can prevent the
+    request class from throwing warnings.
+    """
+
+    def new_function(*args, **kwargs):
+        # raise logging level to CRITICAL
+        logger = logging.getLogger("django.request")
+        previous_logging_level = logger.getEffectiveLevel()
+        logger.setLevel(logging.CRITICAL)
+
+        # trigger original function that would throw warning
+        original_function(*args, **kwargs)
+
+        # lower logging level back to previous
+        logger.setLevel(previous_logging_level)
+
+    return new_function
```

Comment:
I'm not sure about the need for this function, are you able to explain? Even if definitely required, i'm not sure why we're putting either of these functions in the `ciim` app?

Reply:
This function is used in tests to suppress messages like 503 `Service Unavailable: /`, Not Found, to make the test output clean.  Do you suggest any other location to add it?

Reply:
Maybe we can just update the logger config in the test settings? There's a complex example here that overrides several Django loggers: https://docs.djangoproject.com/en/4.0/topics/logging/.